src/valid/interface.rs

```diff
@@ -701,6 +703,28 @@ impl super::Validator {
             bg.clear();
         }
 
+        #[cfg(feature = "validate")]
+        {
+            let mut used_push_constants = module
+                .global_variables
+                .iter()
+                .filter(|&(_, var)| var.space == crate::AddressSpace::PushConstant)
+                .map(|(handle, _)| handle)
+                .filter(|&handle| !info[handle].is_empty());
+            // Check if there is one push constant. Some iterators can restart iteration after
+            // `next` returns None, so don't continue if the iterator returns None.
+            if used_push_constants.next().is_some() {
```

Comment:
Can we instead use [`.skip(1)`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip)? And replace the match below with just a for loop? In the future we'll want to keep accumulating as many errors as possible.

Reply:
That is much cleaner, thank you!

Reply:
Ah actually, I couldn't use a loop else clippy gets upset. I just used `if let` to handle it.

Reply:
NEVERMIND, I just disabled the lint for the never loop temporarily and added a comment about it.